src/components/dialogs/CollaborationRequestDialog.jsx

```diff
@@ -69,36 +78,36 @@ export default function CollaborationRequestDialog({
           display="basic"
           id="DECLINE_REQUEST"
           onClick={async () => {
-            const successful = await patchCollaboration(
-              collaborationId,
-              [
-                {
-                  op: 'replace',
-                  path,
-                  value: 'declined',
-                },
-              ],
-            );
+            collabPatchArgs(collaborationId, [
+              {
+                op: 'replace',
+                path,
+                value: 'declined',
+              },
+            ]);
 
-            if (successful) onCloseDialog();
+            if (isSuccess) {
+              onCloseDialog();
+              queryClient.invalidateQueries(queryKeys.collaborations);
+            }
           }}
         />
         <Button
           loading={loading}
           display="primary"
           onClick={async () => {
-            const successful = await patchCollaboration(
-              collaborationId,
-              [
-                {
-                  op: 'replace',
-                  path,
-                  value: 'approved',
-                },
-              ],
-            );
+            collabPatchArgs(collaborationId, [
+              {
+                op: 'replace',
+                path,
+                value: 'approved',
+              },
+            ]);
 
-            if (successful) onCloseDialog();
+            if (isSuccess) {
```

Comment:
I think that this means that I'm fundamentally misunderstanding react-query in some important way. I don't know how one would go about resolving this. And I suspect that you are right, because this button often requires more than one click to work.

Reply:
Done. For posteriry, we ended up using mutateAsync to resolve this.